<!-- type="str|int|double|bool|yaml"(optional) -->
<launch>
	<arg name="gazebo" 						default="false"/>
	<arg name="ugv_n" 						default="ugv1" />
	<arg name="roi_halfwindow" 		default="100" />
	<arg name="PixelScale" 				default="100.0" />
	<arg name="saveraw" 					default="false" />
	<arg name="saverect" 					default="false" />
	<arg name="trigger_saveraw" 	default="false" />
	<arg name="uav_TF_parent" 		default="$(arg ugv_n)/base_footprint" />
	<arg name="uav_TF_child" 			default="/hast/uav/stereo_raw" />
	<arg name="showcircles" 			default="false"/>
	<arg name="showYUV" 					default="false"/>
	<arg name="showBool" 					default="false"/>
	<arg name="save_images"				default="false"/>
	<arg name="raw_trigger_topic" default="$(arg ugv_n)/pgrstereo/saveTrigger"/>
	<arg name="left_raw_topic" 		default="$(arg ugv_n)/pgrstereo/left/image_raw"/>
	<arg name="right_raw_topic" 	default="$(arg ugv_n)/pgrstereo/right/image_raw"/>
	<arg name="left_rect_topic" 	default="$(arg ugv_n)/pgrstereo/left/image_rect_color"/>
	<arg name="right_rect_topic" 	default="$(arg ugv_n)/pgrstereo/right/image_rect_color"/>


	<arg name="ugvugvsensor_topic"					default="/hast/ugvugvsensor" /> <!-- ugvugvsensor_topic is the client/service to get ugv measurement from vicon -->
	<arg name="ugv_sensor_pose_pub_topic"		default="/hast/$(arg ugv_n)_stereo/ugvugvsensor" /> <!-- publisher to broadcast measurement of ugv -->
	<!-- this is defind within the stereoobs function
	<arg     if="$(arg gazebo)" name="ugv_sensor_frame"			value="/vicon/origin"/>
	<arg unless="$(arg gazebo)" name="ugv_sensor_frame" 		value="/vicon/$(arg ugv_n)/ugv1"/>
	<arg     if="$(arg gazebo)" name="observed_ugv_frame"		value="/vicon/origin"/>
	<arg unless="$(arg gazebo)" name="observed_ugv_frame" 	value="/vicon/ugv1/ugv1"/> -->

	<arg name="color_param_file"			default="$(find hast)/config/stereo_tbrl_params.yaml" />
	<arg name="stereo_param_file"			default="$(find hast)/cam_info/ugvn/$(arg ugv_n)_pgr_params.yaml" />
	<arg name="Hcam2ugv_param_file"		default="$(find hast)/cam_info/ugvn/$(arg ugv_n)_H_cam2ugv.yaml" />

	<!-- <node 	pkg="hast" 	type="stereoObs"	name="$(arg ugv_n)_stereoObs"	output="screen" > -->
	<node 	pkg="hast" 	type="stereoObs"	name="$(arg ugv_n)_stereoObs_$(env HOST)" output="screen" >
		<rosparam command="load" 				file="$(arg color_param_file)" />
		<rosparam command="load" 				file="$(arg stereo_param_file)" />
		<rosparam command="load" 				file="$(arg Hcam2ugv_param_file)" />
		<param name="color_param_file" 			value="$(arg color_param_file)"/>
		<param name="stereo_param_file" 		value="$(arg stereo_param_file)"/>
		<param name="Hcam2ugv_param_file" 	value="$(arg Hcam2ugv_param_file)"/>

		<param name="ugv_n" 							value="$(arg ugv_n)"/>
		<param name="gazebo" 							value="$(arg gazebo)"/>
		<param name="uav_TF_parent" 			value="$(arg uav_TF_parent)"/>
		<param name="uav_TF_child" 				value="$(arg uav_TF_child)"/>
		<param name="saveraw" 						value="$(arg saveraw)"/>
		<param name="saverect" 						value="$(arg saverect)"/>
		<param name="trigger_saveraw"			value="$(arg trigger_saveraw)"/>
		<param name="showcircles" 				value="$(arg showcircles)"/>
		<param name="save_images" 				value="$(arg save_images)"/>
		<param name="showYUV" 						value="$(arg showYUV)"/>
		<param name="showBool" 						value="$(arg showBool)"/>
		<param name="roi_halfwindow" 			value="$(arg roi_halfwindow)" type="int" />
		<param name="PixelScale" 					value="$(arg PixelScale)"  type="double" />
		<param name="raw_trigger_topic" 	value="$(arg raw_trigger_topic)"/>
		<param name="left_raw_topic" 			value="$(arg left_raw_topic)"/>
		<param name="right_raw_topic" 		value="$(arg right_raw_topic)"/>
		<param name="left_rect_topic" 		value="$(arg left_rect_topic)"/>
		<param name="right_rect_topic" 		value="$(arg right_rect_topic)"/>

		<param name="Pose_pub" 					value="/$(arg ugv_n)/stereo/pose"/>
		<param name="OdomSwitch_ser" 		value="/$(arg ugv_n)/service/stereo/OdomSwitch"/>

		<param name="cam_cloud_frameID" 	value="$(arg ugv_n)/hast/stereo_center"/>
		<param name="ugv_cloud_frameID" 	value="$(arg ugv_n)/hast/base_footprint"/>
		<param name="stereoCloudCAM_pub" 	value="/$(arg ugv_n)/stereo/markerCloudCAM"/>
		<param name="stereoCloudUGV_pub" 	value="/$(arg ugv_n)/stereo/markerCloudUGV"/>

		<param name="ugvugvsensor_topic" 					value="$(arg ugvugvsensor_topic)"/>
		<param name="ugv_sensor_pose_pub_topic" 	value="$(arg ugv_sensor_pose_pub_topic)"/>


	</node>
</launch>
